discord/abc.py

```diff
@@ -875,6 +882,12 @@ async def send(self, content=None, *, tts=False, embed=None, file=None,
         else:
             allowed_mentions = state.allowed_mentions and state.allowed_mentions.to_dict()
 
+        if reference is not None:
+            try:
+                reference = reference.to_dict()
```

Comment:
this seems a bit shaky and probably makes sense to check if it's a message or messagereference, and raise if neither

Reply:
Can't import from message.py due to circular reference, would have to use `hasattr` or let the `AttributeError` be raised.

Reply:
There are a few methods I use to resolve circular imports.

1) Shared private attribute. So, for example there'd be a tag property such as `self._can_cast_ref` that can be used without importing anything. I don't exactly like this approach here.
2) Private tag type that can be imported standalone that the types inherit from. Basically the same as #1 except using inheritance and `isinstance` instead of flimsy attributes. See, e.g. `BaseActivity` or `_EmojiTag` for examples in the library.